platform.md

```diff
@@ -1092,15 +1096,30 @@ For more information on build reproducibility see [https://reproducible-builds.o
 
 [run-image]
   reference = "<image reference>"
+  [target]
+  id = "<target identifer>"
+  os = "<OS name>"
+  arch = "<architecture>"
+  variant = "<architecture variant>"
+  [target.distribution]
+  name = "<OS distribution name>"
+  version = "<OS distribution version>"
 
 [build-image]
   reference = "<image reference>"
 ```
 
 Where:
-- `previous-image.reference` MUST be either a digest reference to an image in an OCI registry or the ID of an image in a docker daemon
+- `image.reference` MUST be either a digest reference to an image in an OCI registry or the ID of an image in a docker daemon
+- `image.metadata` MUST be the TOML representation of the layer [metadata label](#iobuildpackslifecyclemetadata-json)
 - `run-image.reference` MUST be either a digest reference to an image in an OCI registry or the ID of an image in a docker daemon
-- `previous-image.metadata` MUST be the TOML representation of the layer [metadata label](#iobuildpackslifecyclemetadata-json)
+- `run-image.target.id` is optional and MUST be the value of the label `io.buildpacks.id`
+- `run-image.target` contains the [target data](#target-data) for the image
+  - If target distribution data is missing, it will be inferred from `/etc/os-release` for Linux images; furthermore, if the image contains the label `io.buildpacks.stack.id` with value `io.buildpacks.stacks.bionic`, the lifecycle SHALL assume the following values:
+    - `run-image.target.os = "linux"`
+    - `run-image.target.arch = "arm64"`
+    - `run-image.target.distribution.name = "ubuntu"`
+    - `run-image.target.distribution.version = "18.04"`
```

Comment:
I assume this means if for some reason if target distribution data is missing + `/etc/os-release` doesn't exist these will be blank? Will we try to do something based on the host system?

Reply:
IIRC checking the os-release file was the best we could think of to do with respect to the host system.  TBH it "shouldn't" usually be blank, and while i'm sure there's cases, even e.g. the smallest "distroless" image (distroless/static) still has that file.

Reply:
but yes,  if that information can't be found anywhere, then those values would be blank.

Reply:
FWIW, from https://www.linux.org/docs/man5/os-release.html:

> The file `/etc/os-release` takes precedence over `/usr/lib/os-release`. Applications should check for the former,
> and exclusively use its data if it exists, and only fall back to `/usr/lib/os-release` if it is missing.
> Applications should not read data from both files at the same time.  `/usr/lib/os-release` is the recommended
> place to store OS release information as part of vendor trees.  `/etc/os-release` should be a relative symlink
> to `/usr/lib/os-release`, to provide compatibility with applications only looking at `/etc`. A relative symlink
> instead of an absolute symlink is necessary to avoid breaking the link in a `chroot` or `initrd` environment such
> as dracut.